src/metabase/api/slack.clj

```diff
@@ -9,16 +9,18 @@
 (defendpoint PUT "/settings"
   "Update Slack related settings. You must be a superuser to do this."
   [:as {{slack-token :slack-token, metabot-enabled :metabot-enabled, :as slack-settings} :body}]
-  {slack-token     [Required NonEmptyString]
-   metabot-enabled [Required]}
+  {metabot-enabled [Required]}
   (check-superuser)
-  (try
-    ;; just check that channels.list doesn't throw an exception (a.k.a. that the token works)
-    (when-not config/is-test?
-      (slack/GET :channels.list, :exclude_archived 1, :token slack-token))
-    (setting/set-many! slack-settings)
-    {:ok true}
-    (catch clojure.lang.ExceptionInfo info
-      {:status 400, :body (ex-data info)})))
+  (if (not slack-token)
+    (setting/set-many! { :slack-token nil, :metabot-enabled false })
```

Comment:
Whoops meant to say you should use `(if-not ...` here instead of `(if (not ...)`. That's the idiomatic way of doing things in Clojure